app/gui/src/controller/ide/desktop.rs

```diff
@@ -164,3 +161,27 @@ impl ManagingProjectAPI for Handle {
         .boxed_local()
     }
 }
+
+/// Select a new name for the project in a form of <suggested_name>_N, where N is a unique sequence
+/// number.
+fn choose_new_project_name(
+    mut existing_names: impl Iterator<Item = String>,
+    suggested_name: &str,
+) -> String {
+    let with_suffix = (1..).map(|i| format!("{}_{}", suggested_name, i));
+    let mut candidates = std::iter::once(suggested_name.to_owned()).chain(with_suffix);
+    // The iterator have no end, so we can safely unwrap.
+    let name = candidates.find(|c| !existing_names.contains(c)).unwrap();
+    capitalize_first(&name)
```

Comment:
This should capitalize each segment first letter, and fix the string if segments does not start with a letter.

Reply:
Also, such normalization could be a function in `ReferentName`.

Reply:
There is no need for such capitalizations, but you're right, the `ProjectName` constructor now does the capitalization of the first letter.